actions/reservation.py

```diff
@@ -27,3 +37,87 @@ def get_actual_after_buffer(self) -> datetime.timedelta:
             after = reservation_unit.actions.get_actual_after_buffer(self.reservation.end)
             buffer_time_after = max(after, buffer_time_after)
         return buffer_time_after
+
+    def to_ical(self, *, site_name: str) -> bytes:
+        language: Lang = (  # type: ignore[assignment]
+            self.reservation.reservee_language or self.reservation.user.get_preferred_language()
+        )
+
+        ical_event = Event()
+        # This should be unique such that if another iCal file is created
+        # for the same reservation, it will be the same as the previous one.
+        uid = f"varaamo.reservation.{self.reservation.pk}@{site_name}"
+        summary = self.get_ical_summary(language=language)
+        description = self.get_ical_description(language=language)
+
+        ical_event.add(name=EventProperty.UID, value=uid)
+        ical_event.add(name=EventProperty.DTSTAMP, value=local_datetime())
+        ical_event.add(name=EventProperty.DTSTART, value=self.reservation.begin.astimezone(DEFAULT_TIMEZONE))
+        ical_event.add(name=EventProperty.DTEND, value=self.reservation.end.astimezone(DEFAULT_TIMEZONE))
+
+        ical_event.add(name=EventProperty.SUMMARY, value=summary)
+        ical_event.add(name=EventProperty.DESCRIPTION, value=description, parameters={"FMTTYPE": "text/html"})
```

Comment:
Should HTML description be in `X-ALT-DESC` or `ALTREP` of `DESCRIPTION` (or does icalendar.Event take care of that)?

Reply:
Cf. https://icalevents.com/4019-ics-feed-generation-with-html/

- https://learn.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxcical/1b69f2ac-8f12-49b9-b91c-b7b44b89b451
- https://learn.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxcical/d7f285da-9c7a-4597-803b-b74193c898a8?redirectedfrom=MSDN

Reply:
This is indeed requried for proper HTML description compatibility with Outlook. I made changes and checked that this now works with Oulook as well.